lazy_dataset/core.py

```diff
@@ -1416,7 +1427,12 @@ def __init__(self, examples, name=None):
         self._keys = tuple(self.examples.keys())
 
     def copy(self, freeze=False):
-        return self.__class__(self.examples, name=self.name)
+        # Use __new__ to not deepcopy attributes
```

Comment:
In the init is no deep copy, hence `# Use __new__ to not deepcopy attributes` is not correct.
The overhead is actually a new tuple with pointers to the keys.

Maybe
`# Use __new__ to avoid a copy of keys`
would be more precise.